tests/test_common.ml

```diff
@@ -197,6 +203,47 @@ let get_iscsi_sr state =
   Lwt.return { state with iscsi_sr = Some (iscsi_sr_ref, iscsi_sr_uuid) }
 
 
+let find_template rpc session_id name =
+  VM.get_all_records rpc session_id >>= fun vms ->
+  let filtered = List.filter (fun (_, record) ->
+      (name = record.API.vM_name_label) &&
+      record.API.vM_is_a_template)
+      vms in
+  match filtered with
+  | [] -> Lwt.return None
+  | (x,_) :: _ -> Lwt.return (Some x)
+
+
+let create_mirage_vm state =
+  let rpc = state.master_rpc in
+  let session_id = state.master_session in
+  find_template rpc session_id "Other install media" >>= fun template_opt ->
+  let template = 
+    match template_opt with
+    | Some vm -> vm
+    | None -> 
+      Printf.fprintf stderr "Failed to find suitable template";
+      failwith "No template"
+  in
+  VM.clone rpc session_id template "mirage" >>= fun vm ->
+  VM.provision rpc session_id vm >>= fun _ ->
+  VM.set_PV_kernel rpc session_id vm "/boot/guest/mir-suspend.xen.gz" >>= fun () ->
+  VM.set_HVM_boot_policy rpc session_id vm "" >>= fun () ->
+  VM.set_memory_limits ~rpc ~session_id ~self:vm ~static_min:meg32 ~static_max:meg32 ~dynamic_min:meg32 ~dynamic_max:meg32 >>= fun () ->
+  Lwt.return ({state with mirage_vm = Some vm}, vm)
+
+let find_or_create_mirage_vm state =
+  let rpc = state.master_rpc in
+  let session_id = state.master_session in
+  VM.get_all_records_where ~rpc ~session_id ~expr:"field \"name__label\"=\"mirage\""
+  >>= fun vms ->
```

Comment:
`>>= function vm::_ -> Lwt.return .., | [] -> create_...`?

Then we can get rid of this `List.hd` even though it is guarded.